sematic/resolvers/silent_resolver.py

```diff
@@ -19,10 +19,21 @@ def _schedule_future(self, future: AbstractFuture) -> None:
     def _run_inline(self, future: AbstractFuture) -> None:
         self._set_future_state(future, FutureState.SCHEDULED)
         try:
+            self._start_inline_execution(future.id)
             value = future.calculator.calculate(**future.resolved_kwargs)
             self._update_future_with_value(future, value)
         except Exception as exception:
             self._handle_future_failure(future, exception)
+        finally:
+            self._end_inline_execution(future.id)
+
+    def _start_inline_execution(self, future_id):
+        """Callback called before an inline execution"""
+        pass
+
+    def _end_inline_execution(self, future_id):
+        """Callback called at the end of an inline execution"""
+        pass
```

Comment:
Can't we add this logic to state machine callbacks instead of new callbacks?
Basically in `_future_did_schedule` and others for the end, we could add the log printout.

Reply:
No, the trouble is twofold. The first reason is much more important than the second:
(1) There are multiple ways the existing callbacks may happen, depending on what occurred with the inline execution. Whether the calculator raised an exception, returned a value, or returned another future will all impact which callbacks get called and in what order. We could shotgun print the end-indicator in all the relevant places but I'd rather have it centralized. Additionally, we want to make sure the end-indicator gets shown, no matter what weird stuff happens with the resolver (i.e. if it errors out in a weird way). If the end-indicator doesn't get printed, the log reader could get confused and add a bunch of stuff to the run's logs that really don't belong there.
(2) The resolver may print log statements about its own operations before we have a chance to add the end-line print to the logs, polluting the inline run's logs with stuff that really belongs to the resolver.